### PR TITLE
Add support for custom static headers

### PR DESCRIPTION
**Pull request by vam-google**
**gax** portion: https://github.com/googleapis/gax-java/pull/432
**gcj** portion: https://github.com/GoogleCloudPlatform/google-cloud-java/pull/2690

This also hardcodes generation version in GeneratorVersionProvider as default value (to make tests easier to run under IDE without need to execute prebuild setps).

### CODE REVIEW

**Comment by vam-google** [comment]
@garrettjonesgoogle PTAL

**Comment by vam-google** [comment]
PTAL

File: src/test/java/com/google/api/codegen/testdata/java/java_library.baseline [hunk 1]
```diff
@@ -7651,6 +7749,10 @@ public class LibraryClientTest {
     Assert.assertEquals(optionalRepeatedFixed32, actualRequest.getOptionalRepeatedFixed32List());
     Assert.assertEquals(optionalRepeatedFixed64, actualRequest.getOptionalRepeatedFixed64List());
     Assert.assertEquals(optionalMap, actualRequest.getOptionalMapMap());
+    Assert.assertTrue(
+        channelProvider.isHeaderSent(
+            ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
+            GaxGrpcProperties.getDefaultApiClientHeaderPattern()));
```
Comment (garrettjonesgoogle) [inline src/test/java/com/google/api/codegen/testdata/java/java_library.baseline]: Do the generated tests pass on api-client-staging?

Comment (vam-google) [reply src/test/java/com/google/api/codegen/testdata/java/java_library.baseline]: Our api-client-staging for java is outdated (broken?). But yes, it works (run artman against latest toolkit & gax fro langauge and commons, then run ./gradlew clean test (excluding others, to make things faster) and it works).

**Comment by vam-google** [comment]
@garrettjonesgoogle PTAL

**Pull request merged**